packages/playwright-test/src/worker/workerMain.ts

```diff
@@ -420,13 +417,13 @@ export class WorkerMain extends ProcessRunner {
 
           // Run "afterEach" hooks, unless we failed at beforeAll stage.
           if (shouldRunAfterEachHooks) {
-            const afterEachError = await testInfo._runAndFailOnError(() => this._runEachHooksForSuites(reversedSuites, 'afterEach', testInfo, afterHooksSlot));
+            const afterEachError = await testInfo._runAndFailOnError(() => this._runEachHooksForSuites(reversedSuites, 'afterEach', testInfo));
             firstAfterHooksError = firstAfterHooksError || afterEachError;
           }
 
           // Teardown test-scoped fixtures. Attribute to 'test' so that users understand
           // they should probably increase the test timeout to fix this issue.
-          await testInfo._timeoutManager.runRunnable({ type: 'test', slot: afterHooksSlot }, async () => {
+          await testInfo._timeoutManager.runRunnable({ type: 'test' }, async () => {
```

Comment:
Why does this runnable, marked as `test`, include afterAll hooks? I think those should be outside.